doc/en/skipping.rst

```diff
@@ -247,9 +247,20 @@ imperatively:
         if not valid_config():
             pytest.xfail("failing configuration (but should work)")
 
-This will unconditionally make ``test_function`` ``XFAIL``. Note that no other code is executed
-after ``pytest.xfail`` call, differently from the marker. That's because it is implemented
-internally by raising a known exception.
+.. code-block:: python
+
+    def test_function2():
+        import slow_module
+
+        if slow_module.slow_function():
+            pytest.xfail("slow_module taking too long")
+
+These two examples illustrate situations where you don't want to check for a condition
+at the module level, which is when a condition would otherwise be evaluated for marks.
+
+They will unconditionally make the two examples ``XFAIL``. Note that no other code is executed
+after a ``pytest.xfail`` call, which is different from a marker. This is because current
```

Comment:
I'd leave the last sentence as-is. I don't think there are any plans to change this? Nor can I think of any other possible implementation. And also it would be a breaking change to change anyway.

Reply:
Updated PR based on feedback